### ISSUE TITLE
Performance Issue: Startup very slow

### ISSUE THREAD

**Issue by stevenleeS0ht** [ipython/ipython#11898]
Python version: 3.7.4
IPython version: 7.6.1

When start ipython from command line. It will take long time.

Python Startup:
```
$ time python -c exit

real	0m0.015s
user	0m0.015s
sys	0m0.000s

```

IPython Startup:
```
$ time ipython -c exit

real	0m0.523s
user	0m0.365s
sys	0m0.039s

```

It is slower than [bpython](https://github.com/bpython/bpython).

I am wondering if it is possible to make optimization.

**Comment by stevenleeS0ht** [comment]
@minrk , @takluyver , @Carreau . Is it possible to improve this issue?

**Comment by minrk** [comment]
profiling would be the way to find out. My guess is that it is the growing number of modules and dependencies, but there may be some easy low-hanging fruit.

**Comment by LucianaMarques** [comment]
I believe we could make a bigger project to massive refactor the codebase and also improve design.

**Comment by AgilentGCMS** [comment]
I found this thread searching for a solution to my slow ipython startup times. My hardware is reasonably fast (28-core Intel Broadwell), and yet ipython takes so much longer than python:
```
$ time python -c exit

real	0m0.052s
user	0m0.018s
sys	0m0.022s

$ time ipython -c exit

real	0m2.542s
user	0m0.667s
sys	0m0.907s
```
I'm not loading any additional modules in ipython, and still!

**Comment by Carreau** [comment]
Yes, we are aware, IPython itself is heavy, and I believe most of the time might be in pygments. 
I unfortunately don't have much allocated  time to spend on IPython so right not work is focus on critical bug fixes.

**Comment by eendebakpt** [comment]
@Carreau @stevenleeS0ht To test the IPython loading time I did the following:
```
# pip install importtime-waterfall tuna

python -X importtime -c "import IPython" 2> i.log
tuna i.log
```
This results in the following overview:

![image](https://user-images.githubusercontent.com/883786/158163262-a3ebd00c-d78d-4542-b455-d4b1bc91e5e9.png)

Import time is spend at various places, with some bigger components (e.g. `prompt_toolkit`, `jedi`).
